bin/lint-staged.js

```diff
@@ -42,9 +42,11 @@ cli.option('--cwd [path]', 'run all tasks in specific directory, instead of the
 
 cli.option('-d, --debug', 'print additional debug information', false)
 
-cli.option(
-  '--diff [string]',
-  'override the default "--staged" flag of "git diff" to get list of files. Implies "--no-stash".'
+cli.addOption(
+  new Option(
+    '--diff [string]',
+    'override the default "--staged" flag of "git diff" to get list of files. Implies "--no-stash".'
+  ).implies({ stash: false })
```

Comment:
Cool, didn't know about `implies()`! Still, since _lint-staged_ can be used via Node.js, bypassing the cli, we should make this fix in `lib/index.js` as well. Maybe we need to change the default value in:

https://github.com/okonet/lint-staged/blob/cf691aa188719d9479ceaeffbffe814594fdb65f/lib/index.js#L78-L79

To something like:

```js
    // Stashing should be disabled by default when the `diff` option is used
    stash = diff === true ? false : undefined,
```

Reply:
I think this part is correct. `diff` is a string and `stash` is a boolean which should never be undefined.
Ideally the commander logic should be extracted out to functions so it can be used both in `bin/lint-staged.js` and `lib/index.js`.

Reply:
Right you are.